gitlabform/processors/project/job_token_scope_processor.py

```diff
@@ -0,0 +1,213 @@
+        job_token_scope = project.job_token_scope.get()
+
+        self._process_limit_access_to_this_project_setting(
+            job_token_config, job_token_scope
+        )
+
+        allowlist_config = job_token_config.get("allowlist", {})
+        verbose(f"configuration allowlist: {allowlist_config}")
+
+        info("Processing Job Token allowlist")
+
+        enforce = allowlist_config.get("enforce", False)
+
+        self._process_groups(
+            job_token_scope, allowlist_config.get("groups", []), enforce
+        )
+
+        self._process_projects(
+            project, job_token_scope, allowlist_config.get("projects", []), enforce
+        )
+
+    @staticmethod
+    def _process_limit_access_to_this_project_setting(
+        configuration: dict, job_token_scope: ProjectJobTokenScope
+    ):
+        limit_access_to_this_project: bool = configuration.get(
+            "limit_access_to_this_project", True
+        )
+
+        if limit_access_to_this_project != job_token_scope.inbound_enabled:
+            info(
+                f"Updating project job token scope to limit access: {limit_access_to_this_project}"
+            )
+            job_token_scope.enabled = limit_access_to_this_project
+            job_token_scope.save()
+
+            # Refresh object in order for groups_allowlist and allowlist to exist on Scope for modification later
+            job_token_scope.refresh()
+        else:
+            info(f"Job Token Scope does not need updating")
+
+    def _process_projects(
+        self,
+        project: Project,
+        job_token_scope: ProjectJobTokenScope,
+        projects_allowlist: List,
+        enforce: bool,
+    ):
+        if not projects_allowlist and enforce:
+            warning(
+                "Process will remove existing projects from allowlist, as none set in configuration"
+            )
+
+        existing_allowlist = job_token_scope.allowlist.list()
+
+        project_ids_to_allow = self._get_target_project_ids_from_config(
+            projects_allowlist
+        )
+
+        if len(project_ids_to_allow) > 0:
+            self._add_projects_to_allowlist(
+                project, job_token_scope, existing_allowlist, project_ids_to_allow
+            )
+            job_token_scope.refresh()
+
+        if enforce:
+            info(
+                "Enforce enabled, removing projects no longer defined in config from allowlist"
+            )
+            self._remove_projects_from_allowlist(
+                project, job_token_scope, existing_allowlist, project_ids_to_allow
+            )
+
+    def _process_groups(
+        self,
+        job_token_scope: ProjectJobTokenScope,
+        groups_allowlist: List,
+        enforce: bool,
+    ):
+        if not groups_allowlist and enforce:
+            warning(
+                "Process will remove existing groups from allowlist, as none set in configuration"
+            )
+
+        existing_allowlist = job_token_scope.groups_allowlist.list()
```

Comment:
I would _prefer_ to use list() since that matches intended functionality and GitLab API, but I do remember something about an open PR whereby list() can be inconsistent in `python-gitlab` @amimas ?